quest/api/datasets.py

```diff
@@ -374,16 +375,30 @@ def open_dataset(dataset, fmt=None, **kwargs):
     m = get_metadata(dataset).get(dataset)
     file_format = m.get('file_format')
     path = m.get('file_path')
+    intake_plugin = m.get('intake_plugin')
+    required = []
+    args = m.get('intake_args')
+    if len(args.strip()):
+        # Get args from json if available
+        required = json.loads(m.get('intake_args'))
+    else:
+        raise ValueError('No intake plugin found')
 
     if path is None:
         raise ValueError('No dataset file found')
 
     if file_format not in list_plugins(static.PluginType.IO):
         raise ValueError('No reader available for: %s' % file_format)
 
-    io = load_plugins(static.PluginType.IO, file_format)[file_format]
-    return io.open(path, fmt=fmt, **kwargs)
+    # Use intake plugin to open
+    if intake_plugin:
+        # New code, with 'intake_plugin' added to the local .db
+        plugin_name = 'open_' + intake_plugin
```

Comment:
This seems somewhat fragile. 
You could look directly in the registry
```
import intake
cls = intake.registry[intake_plugin]
source = cls(*args, **kwargs)
```
or, perhaps better, you could construct either the relavant YAML block or a `intake.catalog.local.LocalCatalogEntry` instance, and have Intake do the lookup for you.

Note that in the Intake world, the driver here could be something like "parquet", but it can also be the fully-qualified class name like "intake_parquet.ParquetSource". Of course, if you have additional constrains within Quest, that's fine.